### PR TITLE
QR Code for invoice implementation

### LINKED ISSUE #46: [BARCODE] implement barcode generator in backend
Customer -> OrderController & OrderGuestController

1. Implement QR-Code-Generator
2. Send generated QR-Code to the invoice through dataEmail object 
` const qrcodeImg = await QRCode.toString(createOrder._id)`
3. Create QR-Code image in invoiceTemplate helper. Image should be big and easy to recognize

### PR DESCRIPTION
**Pull request by fathonitr**
## Overview
Generating a Barcode of the Invoice number, and include it in the email receipt. To help during coupon pickup.

## References
- https://www.npmjs.com/package/qrcode
- https://medium.com/@mohsinogen/how-to-generate-qr-codes-with-node-js-and-express-js-a098f9a38525
- UwU

### CODE REVIEW

File: src/controllers/customer/OrderController.js [hunk 1]
```diff
@@ -151,11 +152,19 @@ class OrderController {
       };
 
       const createOrder = await Order.create([payload], { session });
+      const qrcodeImg = await QRCode.toDataURL(
+        createOrder[0].invoiceNumber.toString(),
+        {
+          version: 2,
+        }
+      );
+      console.log(qrcodeImg);
```
Comment (ilhamuh97) [inline src/controllers/customer/OrderController.js]: Console.log nya jangan lupa diapus ya bang

File: src/controllers/customer/OrderGuestController.js [hunk 2]
```diff
@@ -142,10 +143,19 @@ class OrderController {
 
       const createOrder = await Order.create([payload], { session });
 
+      const qrcodeImg = await QRCode.toDataURL(
+        createOrder[0].invoiceNumber.toString(),
+        {
+          version: 2,
+        }
+      );
+      console.log(qrcodeImg);
```
Comment (ilhamuh97) [inline src/controllers/customer/OrderGuestController.js]: Ini juga

**Comment by ilhamuh97** [comment]
Udah ok sih bang menurut saya, paling invoiceNumber ga butuh toString() lagi karna type invoiceNumber di order model udh string

File: src/controllers/customer/OrderController.js [hunk 3]
```diff
@@ -151,11 +152,18 @@ class OrderController {
       };
 
       const createOrder = await Order.create([payload], { session });
+      const qrcodeImg = await QRCode.toDataURL(
+        createOrder[0].invoiceNumber.toString(),
```
Comment (ilhamuh97) [inline src/controllers/customer/OrderController.js]:
```suggestion
        createOrder[0].invoiceNumber,
```

File: src/controllers/customer/OrderGuestController.js [hunk 4]
```diff
@@ -142,10 +143,18 @@ class OrderController {
 
       const createOrder = await Order.create([payload], { session });
 
+      const qrcodeImg = await QRCode.toDataURL(
+        createOrder[0].invoiceNumber.toString(),
```
Comment (ilhamuh97) [inline src/controllers/customer/OrderGuestController.js]:
```suggestion
        createOrder[0].invoiceNumber,
```

File: src/controllers/admin/OrderController.js [hunk 5]
```diff
@@ -125,10 +126,18 @@ class OrderController {
         throw { name: 'Bad Request', message: 'Payment type not found' };
       }
 
+      const qrcodeImg = await QRCode.toDataURL(
+        findUpdatedOrder.invoiceNumber.toString(),
```
Comment (ilhamuh97) [inline src/controllers/admin/OrderController.js]:
```suggestion
        findUpdatedOrder.invoiceNumber,
```

File: src/controllers/customer/OrderController.js (hunk 1 above)
Comment (fathonitr) [reply src/controllers/customer/OrderController.js]: ok  oce

File: src/controllers/customer/OrderController.js (hunk 3 above)
Comment (fathonitr) [reply src/controllers/customer/OrderController.js]: hmm ner bener

File: src/controllers/customer/OrderGuestController.js (hunk 4 above)
Comment (fathonitr) [reply src/controllers/customer/OrderGuestController.js]: oh iya waktu itu kan karena kurang [0] nya ya, bukan karena object type.

Comment (ilhamuh97) [reply src/controllers/customer/OrderGuestController.js]: Iya, kurang [0] karna return nya dalam bentuk array

**Pull request merged**